torch/csrc/jit/passes/bailout_graph.cpp

```diff
@@ -208,15 +244,25 @@ struct BailOutInserter {
         const auto& live_inputs = liveness_sets_[*it];
 
         // guarded inputs come first
-        // currently, there's always one  guaded input
+        // currently, there's always one guaded input
         bailout_node->addInput(it->input());
 
+        // pass loop counts in the same order
+        // a bailout graph expects them
+        auto loopCounts = collectLoopCounts(*it);
```

Comment:
I still do not follow why this is necessary. It seems so fragile.

Reply:
I think it's just the comment is confusing. BailOuts need loop counts and liveness analysis doesn't capture them since they are often not used explicitly. I think it will be a bigger mess to fix this in liveness both logically since we will have to make a value that's never used live (confusing) and logistically since we will need to special case.